### PR TITLE
refactor: project architecture

### LINKED ISSUE #23: feat: support IE 11
**Feature request**

Make possible to support IE 11 (CSS Custom Properties).

**Alternatives for the moment:**
- https://github.com/nuxodin/ie11CustomProperties

### PR DESCRIPTION
**Pull request by lauthieb**
## Pull Request checklist

🚨 Please review the [guidelines for contributing](../blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Don't request your main!
- [x] Make sure you are making a pull request against the **main branch** (left side).
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Description

This refactoring is there in order to anticipate the future arrival of Web Components. For them, we need to consume standalone CSS files. 

So we now introduce two different ways of consumption:
- One full CSS package `@decathlon/vitamin-css` built with TailwindCSS and purgeable by the consumer in order to optimize its production build
- Several standalone component's CSS packages (example: `@decathlon/vitamin-css-button` to only get a CSS standalone file without CSS Custom Properties that only contains button component's styles). So this can also close #23. _(FYI @crotru)_

## Thanks Joffrey! 

**This Pull Request was made in peer programming with @jleveugle** 

## Other information

0 warning :tada:

image

### CODE REVIEW

**Review by thollander — APPROVED** [review]
Seems good to me to setup the architecture

Still some debates to have on the stencil part that is already planned in this PR but I think it can be merged like this.

**Comment by lauthieb** [comment]
Thanks a lot, @thollander for your time and advice!

File: packages/showcases/css/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+  stories: [
+    '../stories/*.stories.(mdx|js)',
+    '../stories/guidelines/**/*.stories.(mdx|js)',
+    '../stories/components/**/*.stories.(mdx|js)',
```
Comment (gabbloquet) [inline packages/showcases/css/.storybook/main.js:5]: I imagine that's not a good practice to put that ? 
```suggestion
    '../stories/**/*.stories.(mdx|js)',
```

**Review by gabbloquet — APPROVED** [review]
Incredible work, look good to me 🚀
Congratulations !

File: packages/showcases/css/.storybook/main.js (hunk 1 above)
Comment (lauthieb) [reply packages/showcases/css/.storybook/main.js:5]: I've separated because we need to see introduction, then guidelines, then components in this order :)

**Comment by lauthieb** [comment]
Thanks @gabbloquet for your review!!

File: packages/showcases/css/.storybook/main.js (hunk 1 above)
Comment (gabbloquet) [reply packages/showcases/css/.storybook/main.js:5]: Great thanks a lot 🚀

**Review by gabbloquet — APPROVED** [review]

**Review by jleveugle — APPROVED** [review]

**Pull request merged**